### PR TITLE
Add more information to job process failure error

### PR DESCRIPTION
**Pull request by rjmholt**
## PR Summary

Adds more information to the exception thrown when the process running a job exits unexpectedly. This is part of an attempt to work out why a particular `Stop-Job` test is failing in CI. See https://github.com/PowerShell/PowerShell/issues/7249.

We can change the particular implementation and error message as needed. Even if we decide not to merge it, opening the PR may still help us to diagnose the problem.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [x] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs [hunk 1]
```diff
@@ -708,8 +708,26 @@ protected void OnExited(object sender, EventArgs e)
...
+            try
+            {
+                var jobProcess = (Process)sender;
+                exitCode = jobProcess.ExitCode;
+                exitMessage = exitCode == 0 ? jobProcess.StandardOutput.ReadToEnd() : jobProcess.StandardError.ReadToEnd();
+            }
+            catch (Exception exception)
+            {
+                exitCode = -1;
+                exitMessage = StringUtil.Format("<Unable to read streams from job process for reason '{0}'>", exception.Message);
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs]: This error message string should also go into the resx file for eventual localization.

File: src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs [hunk 2]
```diff
@@ -708,8 +708,26 @@ protected void OnExited(object sender, EventArgs e)
...
-            PSRemotingTransportException psrte = new PSRemotingTransportException(PSRemotingErrorId.IPCServerProcessExited,
-                                RemotingErrorIdStrings.IPCServerProcessExited);
+
+            int exitCode;
+            string exitMessage;
+            try
+            {
+                var jobProcess = (Process)sender;
+                exitCode = jobProcess.ExitCode;
+                exitMessage = exitCode == 0 ? jobProcess.StandardOutput.ReadToEnd() : jobProcess.StandardError.ReadToEnd();
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs]: Is it possible for both stdout and stderror to have information on exit?  We may want to read both, but I think this is fine for now.

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs (hunk 2 above)
Comment (rjmholt) [reply src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs]: Yeah I was thinking the same - wasn't sure what the policy is on cramming too much information into a single error message

File: src/System.Management.Automation/resources/RemotingErrorIdStrings.resx [hunk 3]
```diff
@@ -1657,4 +1657,10 @@ All WinRM sessions connected to PowerShell session configurations, such as Micro
   <data name="WSManClientDllNotAvailable" xml:space="preserve">
     <value>This parameter set requires WSMan, and no supported WSMan client library was found. WSMan is either not installed or unavailable for this system.</value>
   </data>
+  <data name="ProcessExitInfo" xml:space="preserve">
+    <value>Exit code {0}, stdout '{1}', stderr '{2}'.</value>
```
[the remark this answers is not in the document]
Comment (iSazonov) [reply src/System.Management.Automation/resources/RemotingErrorIdStrings.resx]: We already have multiline error messages.

**Review by PaulHigin — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
Reopen the PR to restart CIs.

**Pull request merged**